Lib/fractions.py

```diff
@@ -594,8 +594,18 @@ def __abs__(a):
         """abs(a)"""
         return Fraction(abs(a._numerator), a._denominator, _normalize=False)
 
+    def __int__(a):
+        """int(a)"""
+        if a._numerator < 0:
+            return int(-(-a._numerator // a._denominator))
```

Comment:
Would not be better to use `operator.index()`?

Reply:
I'm not sure why; `int` seems more natural here, given that we're implementing `__int__`, and it seems a little bit more obvious to me that `int` *must* return something of type `int` (compared to `__index__`). I think it doesn't really matter all that much - anything implementing `Integral` and returning different results from `__int__` and `__index__` is asking for trouble.

Also, in `numbers.Integral`, `__int__` is the more fundamental method: `__index__` is implemented in terms of `__int__` by default.